geoengine/ml.py

```diff
@@ -115,18 +118,66 @@ def check_data_type(data_type: TypeProto, expected_type: RasterDataType, prefix:
         raise InputException('Models with multiple inputs are not supported')
     check_data_type(model_inputs[0].type, input_type, 'input')
 
-    dims = model_inputs[0].type.tensor_type.shape.dim
-    if len(dims) != 2:
-        raise InputException('Only 2D input tensors are supported')
-    if not dims[1].dim_value:
-        raise InputException('Dimension 1 of the input tensor must have a length')
-    if dims[1].dim_value != num_input_bands:
-        raise InputException(f'Model input has {dims[1].dim_value} bands, but {num_input_bands} bands are expected')
+    dim = model_inputs[0].type.tensor_type.shape.dim
```

Comment:
can we not just outsource the validation to the backend?

Reply:
i simplified it and now we have a method that is also usable for metadata definition